frontend/src/Components/ContentComponent.tsx

```diff
@@ -128,6 +134,31 @@ function processVideoEmbed(img: HTMLImageElement) {
     return !!videoUrl;
 }
 
+function updateVimeo(iframe: HTMLIFrameElement) {
+
+    loadVimeoPlayer(iframe);
+    observeOnHidden(iframe, () => stopVideo(iframe));
+
+}
+
+function loadVimeoPlayer(iframe: HTMLIFrameElement) {
+    const attachVimeoPlayer = () => {
+        const player = new window.Vimeo.Player(iframe);
+        player.on('play', function() {
+            stopInnerVideos(document.body, iframe);
+        });
+    };
+
+    if (window.Vimeo) {
+        attachVimeoPlayer();
+    } else {
+        const tag = document.createElement('script');
```

Comment:
to prevent duplicate script inclusion in slow network situation
```suggestion
    } else if(!document.getElementById('vimeo-script')){
        const tag = document.createElement('script');
        tag.id = 'vimeo-script';
```